cli/popper/commands/cmd_run.py

```diff
@@ -226,7 +240,8 @@ def run_workflow(**kwargs):
     # Initialize a Workflow. During initialization all the validation
     # takes place automatically.
     wf = Workflow(kwargs['wfile'])
-    wf_runner = WorkflowRunner(wf)
+    wf_runner = WorkflowRunner(wf, kwargs['substitutions'],
+                               kwargs['allow_loose'])
```

Comment:
since substitutions are relevant to workflow parsing (not execution), there is no need for the runner to be aware of these options. Instead, these can be arguments to the `Workflow` constructor (line 242 above)

Reply:
Fixed.